### PR TITLE
Return only unique names/versions for LoggerCollection

### LINKED ISSUE #10954: LoggerCollection name redundancy
## 🐛 Bug

When you create a logger collection, the name of the logger collection is used for the save path in `ModelCheckpoint`. LoggerCollection concats  the names of all logger. Therefore, in my example the checkpoints are saved to `f"{experiment_name}_{experiment_name}"` instead of `experiment_name`.

### To Reproduce

```python
tb_logger = TensorBoardLogger(save_dir=root_dir, name=experiment_name, version=version)
csv_logger = CSVLogger(save_dir=root_dir, name=experiment_name, version=version)
trainer = pl.Trainer(
        logger=[
            csv_logger,
            tb_logger,
        ],
    )

#...

class TestModule(pl.LightningModule):
    def configure_callbacks(self):
        checkpoint = ModelCheckpoint(
            monitor="val_map",
            filename="{epoch}-{step}--{val_map:.4f}",
            auto_insert_metric_name=True,
            every_n_epochs=1,
        )
        return [checkpoint]
```

### Expected behavior

I would expect the checkpoints to be saved under the same name I set for the loggers. I understand the motive behind the current implementation, but I think only unique names should be concatenated there. Same goes for the version property.

### Environment

- PyTorch Lightning Version (e.g., 1.5.0): 1.5.4
- PyTorch Version (e.g., 1.10): 1.10
- Python version (e.g., 3.9): 3.8
- OS (e.g., Linux): Linux
- CUDA/cuDNN version:
- GPU models and configuration: 1080
- How you installed PyTorch (`conda`, `pip`, source): pip
- If compiling from source, the output of `torch.__config__.show()`: -
- Any other relevant information:

### Additional context

Workaround is to set the `dirpath` argument manually.

cc @borda @awaelchli @edward-io @ananthsub @tchaton

### PR DESCRIPTION
**Pull request by twsl**
## What does this PR do?

Fixes #10954

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by tchaton — APPROVED** [review]
LGTM !

File: pytorch_lightning/loggers/base.py [hunk 1]
```diff
@@ -452,13 +452,15 @@ def save_dir(self) -> Optional[str]:
 
     @property
     def name(self) -> str:
-        """Returns the experiment names for all the loggers in the logger collection joined by an underscore."""
-        return "_".join(str(logger.name) for logger in self._logger_iterable)
+        """Returns the unique experiment names for all the loggers in the logger collection joined by an
+        underscore."""
+        return "_".join(list(dict.fromkeys(str(logger.name) for logger in self._logger_iterable)))
```
Comment (rohitgr7) [inline pytorch_lightning/loggers/base.py:457]:
```suggestion
        return "_".join(dict.fromkeys(str(logger.name) for logger in self._logger_iterable))
```

File: pytorch_lightning/loggers/base.py [hunk 2]
```diff
@@ -452,13 +452,15 @@ def save_dir(self) -> Optional[str]:
...
 
     @property
     def version(self) -> str:
-        """Returns the experiment versions for all the loggers in the logger collection joined by an underscore."""
-        return "_".join(str(logger.version) for logger in self._logger_iterable)
+        """Returns the unique experiment versions for all the loggers in the logger collection joined by an
+        underscore."""
+        return "_".join(list(dict.fromkeys(str(logger.version) for logger in self._logger_iterable)))
```
Comment (rohitgr7) [inline pytorch_lightning/loggers/base.py:463]:
```suggestion
        return "_".join(dict.fromkeys(str(logger.version) for logger in self._logger_iterable))
```

**Review by rohitgr7 — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]
Strictly speaking, this is a breaking change. I labelled it with milestone 1.6 for now. I would be careful to include this in the 1.5.x release since this impacts the path where artifacts are stored, so it may also impact resuming for some users.

File: tests/loggers/test_base.py [hunk 3]
```diff
@@ -56,17 +56,62 @@ def test_logger_collection():
...
+
+def test_logger_collection_names_order():
+    logger1 = CustomLogger(name="name1")
+    logger2 = CustomLogger(name="name2")
+    logger3 = CustomLogger(name="name1")
+    logger4 = CustomLogger(name="name3")
+
+    logger = LoggerCollection([logger1, logger2, logger3, logger4])
+
+    assert logger.name == f"{logger1.name}_{logger2.name}_{logger4.name}"
```
Comment (Borda) [inline tests/loggers/test_base.py:77]:
```suggestion
    loggers = [CustomLogger(name=n) for n in ("name1", "name2", "name1", "name3")
    logger = LoggerCollection(loggers)
    assert logger.name == f"{loggers[0].name}_{loggers[1].name}_{loggers[3].name}"
```

File: tests/loggers/test_base.py [hunk 4]
```diff
@@ -56,17 +56,62 @@ def test_logger_collection():
...
+
+def test_logger_collection_versions_order():
+    logger1 = CustomLogger(version="1")
+    logger2 = CustomLogger(version="2")
+    logger3 = CustomLogger(version="1")
+    logger4 = CustomLogger(version="3")
+
+    logger = LoggerCollection([logger1, logger2, logger3, logger4])
+
+    assert logger.version == f"{logger1.version}_{logger2.version}_{logger4.version}"
```
Comment (Borda) [inline tests/loggers/test_base.py:98]:
```suggestion
    loggers = [CustomLogger(version=v) for v in ("1", "2", "1", "3")]
    logger = LoggerCollection(loggers)
    assert logger.version == f"{loggers[0].version}_{loggers[1].version}_{loggers[3].version}"
```

**Review by Borda — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]
great!!! happy holidays
